src/resources/HttpResource.ts

```diff
@@ -13,20 +15,21 @@ export type HttpResourceOptions = {
 };
 
 export default class HttpResource<
-  TApi extends HttpApi = HttpApi
-> extends Resource<HttpContext<TApi>> {
+  TApi extends HttpApi = HttpApi,
+  TContext extends HttpContext<TApi> = HttpContext<TApi>
+> extends Resource<TContext> {
   readonly api: TApi;
 
   _endpoint: Endpoint;
 
-  constructor(context: HttpContext<TApi>, { endpoint }: HttpResourceOptions) {
+  constructor(context: TContext, { endpoint }: HttpResourceOptions) {
     super(context);
 
     this.api = context.httpApi;
     this._endpoint = endpoint;
   }
 
-  get<T>(id?: string): (T | null | undefined) | Promise<T | null | undefined> {
```

Comment:
should this no longer be parametrized on `<T = Obj>`?

is that just useless because TS doesn't do this sort of inference? if so, then why do we keep this on `HttpApi`?

Reply:
we can add it, however it's just not a lot of value add. I am pretty confident we could drop it from httpApi, but I view it as lower level so it might be good to give the option to specify something